### DISCUSSION TITLE
Nuxi upgrade question

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [nuxt/nuxt#27352] [Questions] [answered]
💚👋

I have a question for a long time now about running `nuxi upgrade -f`. As per the recommendation, I always run it after creating a new app.
But when I have to reinstall deps, for example now that I'm trying Nightly for Nuxt 4 stuff, 
do we have to call it after removing the lock file and reinstalling the deps?

So:

1. Remove lock file
2. Change nuxt to nightly in package.json
3. pnpm install
4. pnpx nuxi upgrade -f

Or just:

1. Remove lock file
2. Change nuxt to nightly in package.json
3. pnpm install

Will appreciate the info.

**Comment by xjccc** [comment, marked as answer]
When you execute `pnpx nuxi upgrade -f`, it will recreating `node_modules` and `pnpm-lock`, then install latest `nuxt 3`
![image](https://github.com/nuxt/nuxt/assets/16478776/df0e5110-df9f-47f3-808d-2970e59d613e)
